kurtosis-devnet/tests/isthmus/withdrawal_root_test.go

```diff
@@ -0,0 +1,124 @@
+package isthmus
+
+import (
+	"bytes"
+	"math/big"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ethereum/go-ethereum/common"
+	gtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+
+	"github.com/ethereum-optimism/optimism/devnet-sdk/contracts/constants"
+	"github.com/ethereum-optimism/optimism/devnet-sdk/system"
+	"github.com/ethereum-optimism/optimism/devnet-sdk/testing/systest"
+	"github.com/ethereum-optimism/optimism/devnet-sdk/types"
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils/wait"
+	"github.com/ethereum-optimism/optimism/op-service/client"
+	"github.com/ethereum-optimism/optimism/op-service/predeploys"
+	"github.com/ethereum-optimism/optimism/op-service/solabi"
+	"github.com/ethereum-optimism/optimism/op-service/sources"
+	"github.com/ethereum-optimism/optimism/op-service/testlog"
+)
+
+func withdrawalRootTestScenario(chainIdx uint64, userSentinel interface{}) systest.SystemTestFunc {
+	return func(t systest.T, sys system.System) {
+		ctx := t.Context()
+
+		chain := sys.L2(chainIdx)
+		logger := testlog.Logger(t, log.LevelInfo)
+		logger.Info("Started test")
+
+		user := ctx.Value(userSentinel).(types.Wallet)
+
+		// Sad eth clients
+		rpcCl, err := client.NewRPC(ctx, logger, chain.RPCURL())
+		require.NoError(t, err)
+		t.Cleanup(rpcCl.Close)
+		ethCl, err := sources.NewEthClient(rpcCl, logger, nil, sources.DefaultEthClientConfig(10))
+		require.NoError(t, err)
+
+		gethCl, err := ethclient.DialContext(ctx, chain.RPCURL())
+		require.NoError(t, err)
+		t.Cleanup(gethCl.Close)
+
+		// Determine pre-state
+		preBlock, err := gethCl.BlockByNumber(ctx, nil)
+		require.NoError(t, err)
+		logger.Info("Got pre-state block", "hash", preBlock.Hash(), "number", preBlock.Number())
+
+		preBlockHash := preBlock.Hash()
+		preProof, err := ethCl.GetProof(ctx, predeploys.L2ToL1MessagePasserAddr, nil, preBlockHash.String())
+		require.NoError(t, err)
+		preWithdrawalsRoot := preProof.StorageHash
+
+		// check isthmus withdrawals-root in the block matches the state
+		gotPre := preBlock.WithdrawalsRoot()
+		require.NotNil(t, gotPre)
+		require.Equal(t, preWithdrawalsRoot, *gotPre, "withdrawals root in block is what we expect")
+
+		chainID := (*big.Int)(chain.ID())
+		signer := gtypes.LatestSignerForChainID(chainID)
+		priv, err := crypto.HexToECDSA(user.PrivateKey())
+		require.NoError(t, err)
+
+		// construct call input, ugly but no bindings...
```

Comment:
should we open an issue for writing these bindings? @sigma

Reply:
I can take this https://github.com/ethereum-optimism/optimism/issues/14473

Reply:
I recommend just using [w3](https://github.com/lmittmann/w3) and not doing anything too fancy here